toolchain/check/check_unit.cpp

```diff
@@ -325,6 +334,53 @@ auto CheckUnit::ImportOtherPackages(SemIR::TypeId namespace_type_id) -> void {
   }
 }
 
+auto CheckUnit::ImportCppPackages() -> void {
+  const auto& imports = unit_and_imports_->cpp_imports;
+  if (imports.empty()) {
+    return;
+  }
+
+  if (imports.size() >= 2) {
+    // TODO: Support multiple Cpp imports.
+    CARBON_DIAGNOSTIC(CppInteropMultipleImports, Error,
+                      "multiple ({0}) Cpp imports are not supported", int);
+    emitter_.Emit(imports[1].node_id, CppInteropMultipleImports,
+                  imports.size());
+    return;
+  }
+
+  const auto& import = imports.front();
+  llvm::StringRef cpp_file_path =
+      unit_and_imports_->unit->value_stores->string_literal_values().Get(
+          import.library_id);
+
+  auto file = fs_->openFileForRead(cpp_file_path);
+  if (!file) {
+    CARBON_DIAGNOSTIC(CppInteropCantReadFile, Error,
+                      "file '{0}' couldn't be opened for reading", std::string);
+    CARBON_DIAGNOSTIC(CppInteropCantReadFileNote, Note, "File read error: {0}",
+                      std::string);
+    emitter_.Build(import.node_id, CppInteropCantReadFile, cpp_file_path.str())
+        .Note(import.node_id, CppInteropCantReadFileNote,
+              file.getError().message())
+        .Emit();
+    return;
+  }
+
+  llvm::vfs::File& file_ref = *file.get();
+  auto buffer = file_ref.getBuffer(cpp_file_path);
+  if (!buffer) {
+    CARBON_DIAGNOSTIC(CppInteropFailedAccessingFileBuffer, Error,
+                      "failed accessing buffer of file '{0}': {1}", std::string,
+                      std::string);
+    emitter_.Emit(import.node_id, CppInteropFailedAccessingFileBuffer,
+                  cpp_file_path.str(), buffer.getError().message());
+    return;
+  }
```

Comment:
I was expecting that you'd go the path of generating a `#include` file and letting Clang take care of these diagnostics. Given you're trying to avoid that for now, can you please take a look at whether you can use SourceBuffer::MakeFromFile here?

Reply:
I'm using `SourceBuffer::MakeFromFile()` now, this definitely helps with avoiding code duplication.
A couple of notes:
1. From the documentation, `SourceBuffer` should be used for carbon files, and I only guess the documentation just needs to be updated.
2. `SourceBuffer` doesn't allow propagating the location where the filename was referenced so that diagnostics would point to it. I have some ideas on how to add support for that, but for now just added a TODO.